library/index.c

```diff
@@ -530,10 +530,14 @@ int eblob_generate_sorted_index(struct eblob_backend *b, struct eblob_base_ctl *
 	struct eblob_map_fd src, dst;
 	int fd, err, len;
 	char *file, *dst_file;
+	int cache_is_empty = 0;
 
 	if (b == NULL || bctl == NULL)
 		return -EINVAL;
 
+	/* check that eblob cache is empty. If cache is empty we can skip iterating and flushing it */
+	cache_is_empty = eblob_cache_empty(b);
+
```

Comment:
> > Everything is a matter of seconds, even unixtime is number of seconds from 1 January 1970.
> This is not a matter of seconds.

In our case it tooks a lot of time which is not a matter of seconds.
> As I already said, it linearly depends on a number of keys in blob. And this dependence can dramatically increase time of eblob initialization.
Let's suppose these are serious time wasters, in this case this problem should be fixed without introduction of binlog mess. I showed multiple times the smallest change which addresses this issue, I will implement it.

We are going past each other, Kirill, that's enough. It would be great if you set a test environment with parallel write of the big files and simultaneous sorting process to figure out what causes the problem. If you do not think it is a binary log issue, then you can try to catch it up in older version like 0.22.18 or 0.22.16.